lib/Controller/ApiController.php

```diff
@@ -126,17 +132,29 @@ public function getRooms() {
 	 * @return JSONResponse
 	 */
 	public function createRoom($roomName) {
-		$qb = $this->dbConnection->getQueryBuilder();
-		$qb->insert('spreedme_rooms')
+		$query = $this->dbConnection->getQueryBuilder();
+		$query->insert('spreedme_rooms')
 			->values(
 				[
-					'name' => $qb->createNamedParameter($roomName),
+					'name' => $query->createNamedParameter($roomName),
 				]
-			)->execute();
+			);
+
+		try {
+			$query->execute();
+		} catch (UniqueConstraintViolationException $e) {
+			return new JSONResponse(
+				[
+					'status' => 'error',
+					'message' => $this->l10n->t('A room with this name already exists.'),
+				]
+			);
+		}
 
 		return new JSONResponse(
 			[
-				'roomId' => $qb->getLastInsertId(),
+				'status' => 'success',
```

Comment:
What about using another statuscode than 200 and matching on .error in jQuery then? :)